documentation/src/main/asciidoc/topics/upgrading.adoc

```diff
@@ -123,6 +123,12 @@ Any annotations will simply be copied as comments to the generated `.proto` file
 
 The `filesystem-realm`, an experimental feature which has never been documented, has been removed.
 
+== Query efficiency
+
+Restore the default query max results that used to be present in older Infinispan versions (prior of 12.1.5.Final)
+and provide a new configuration attribute to change this default, that is now restored to 100.
+Limiting the number of results returned by a query. Applies to indexed, non-indexed, and hybrid queries.
+It significantly improves performance of queries that don't have an explicit limit set.
```

Comment:
```suggestion
== Query efficiency
The `default-max-results` property that was present in Infinispan 12.1.5.Final and earlier versions has been restored.
This property limits the number of results returned by a query and significantly improves performance of queries that don't have an explicit limit set.
The `default-max-results` value is set to 100.
```

Reply:
Not quite @dvagnero : Infinispan never had a `default-max-results` property. It just had a default, unmodifiable value for it.

Reply:
Thanks @dvagnero, I merged them:

> The default query max results that was present in Infinispan 12.1.5.Final and earlier versions has been restored,
together with the fact that you can now also change this default by setting the `query.default-max-results` cache property.
Limiting the number of results returned by a query significantly improves performance of queries that don't have an explicit limit set.
The default value of `default-max-results` is 100.

@dvagnero what do you think of?

Reply:
Thanks! What about...
The default for maximum results returned by a query that was present in Infinispan 12.1.5.Final and earlier versions has been restored. You can now change the default limit by setting the `default-max-results` cache property. The default value of `default-max-results` is 100.
Limiting the number of results returned by a query significantly improves performance of queries that don't have an explicit limit set.

Reply:
sounds great to me... I'm applying it. Thanks